File: legacy/OrderMatchingEngine/src/main/java/com/ndgndg91/ordermatchingengine/order/dto/request/CancelOrderRequest.java
package com.ndgndg91.ordermatchingengine.order.dto.request;

import com.ndgndg91.ordermatchingengine.order.OrderType;
import com.ndgndg91.ordermatchingengine.order.Symbol;
import com.ndgndg91.ordermatchingengine.order.validation.ValueOfEnum;
import lombok.Getter;
import lombok.ToString;

import javax.validation.constraints.NotNull;

@Getter
@ToString
public final class CancelOrderRequest {
    @NotNull(message = "order id is required")
    private String orderId;
    @ValueOfEnum(enumClass = OrderType.class)
    @NotNull(message = "order type is required")
    private String orderType;
    @ValueOfEnum(enumClass = Symbol.class)
    @NotNull(message = "symbol is required")
    private String symbol;
}

File: OrderMatchingEngineKotlin/src/main/kotlin/com/ndgndg91/ordermatchingenginekotlin/order/dto/request/CancelOrderRequest.kt
package com.ndgndg91.ordermatchingenginekotlin.order.dto.request

import com.ndgndg91.ordermatchingenginekotlin.order.validation.OrderTypeValue
import com.ndgndg91.ordermatchingenginekotlin.order.validation.SymbolValue
import javax.validation.constraints.NotNull

data class CancelOrderRequest(
    @field:NotNull(message = "order id is required") val orderId: String?,
    @field:OrderTypeValue @field:NotNull(message = "order type is required") val orderType: String?,
    @field:SymbolValue @field:NotNull(message = "symbol is required") val symbol: String?
)
